### PR TITLE
Remove Python dependency six -- again

### PR DESCRIPTION
**Pull request by cclauss**
Closes #6560 with the help of `make git`

Remove Python 2 / 3 compatibility dependency [`six`](https://pypi.org/project/six)

Leave the dependency in `requirements.txt` for vendored code such as Infogami.

### CODE REVIEW

File: openlibrary/catalog/utils/__init__.py [hunk 1]
```diff
@@ -179,7 +174,7 @@ def match_with_bad_chars(a, b):
         return True
 
     def drop(s):
-        return re_drop.sub('', six.ensure_text(s))
+        return re_drop.sub('', s.decode() if isinstance(s, bytes) else s)
```
Comment (cdrini) [inline openlibrary/catalog/utils/__init__.py:177]: This one looks a touch confusing; can you expand on what's happening?

**Review by cdrini — APPROVED** [review]
Woohoo!! This looks great :blush: Code lgtm ; one comment. Let me try putting this on testing

File: openlibrary/catalog/utils/__init__.py (hunk 1 above)
Comment (cclauss) [reply openlibrary/catalog/utils/__init__.py:177]: https://github.com/benjaminp/six/blob/master/six.py#L939

Comment (cdrini) [reply openlibrary/catalog/utils/__init__.py:177]: Lgtm! It looks like it's using the default decode params there as well. Only difference is this won't error if `s` is non-str AND non-bytes, but that's probably ok?

**Comment by cdrini** [comment]
Testing looks stable; ship it!

**Pull request merged**